src/identities/providers/github/GitHubProvider.ts

```diff
@@ -280,102 +280,92 @@ class GitHubProvider extends Provider {
-        break;
-      } else {
-        pageNum = pageNum + 1;
-      }
-    }
-    pageNum = 1;
-    while (true) {
-      const request = this.createRequest(
-        `${this.apiUrl}/user/followers?per_page=100&page=${pageNum}`,
-        {
-          method: 'GET',
-        },
-        providerToken,
-      );
-      const response = await fetch(request);
-      if (!response.ok) {
-        if (response.status === 401) {
-          throw new identitiesErrors.ErrorProviderUnauthenticated(
-            `Invalid access token`,
+        const error = data?.errors?.at?.(0);
+        if (error != null) {
+          throw new identitiesErrors.ErrorProviderCall(
+            `Provider response body contains an error: ${error.message}`,
+            {
+              data: error
+            }
           );
         }
-        throw new identitiesErrors.ErrorProviderCall(
-          `Provider responded with ${response.status} ${response.statusText}`,
-        );
-      }
-      let data;
-      try {
-        data = await response.json();
-      } catch (e) {
-        throw new identitiesErrors.ErrorProviderCall(
-          `Provider response body is not valid JSON`,
-          { cause: e },
-        );
-      }
-      for (const item of data) {
-        const identityData = await this.getIdentityData(
-          authIdentityId,
-          item.login,
-        );
-        if (
-          identityData &&
-          identitiesUtils.matchIdentityData(identityData, searchTerms)
-        ) {
-          yield identityData;
+        const foundIdentityData: any[] = data?.data?.user?.[identityGroup]?.nodes ?? [];
+        for (const identityData of foundIdentityData) {
+          identityData.providerId = this.id;
+          if (!foundIdentityIds.has(identityData.identityId) && identitiesUtils.matchIdentityData(identityData, searchTerms)) {
+            foundIdentityIds.add(identityData.identityId);
+            yield identityData;
+          }
+        }
+        if (foundIdentityData.length === 0) {
+          break;
+        } else {
+          const endCursor: string | undefined = data?.data?.user?.followers?.pageInfo.endCursor;
+          if (endCursor == null) break;
+          cursor = endCursor;
         }
-      }
-      if (data.length === 0) {
-        break;
-      } else {
-        pageNum = pageNum + 1;
       }
     }
   }
 
+  protected getConnectedIdentityDatasRequestBody(
+    authIdentityId: IdentityId,
+    identityGroup: 'following' | 'followers',
+    cursor?: string
+  ): string {
+    const query = `query {
+      user(login: "${authIdentityId}") {
+        ${identityGroup}(first: 100${cursor == null ? '' : `, after: "${cursor}"`}) {
+          nodes {
+            identityId: login
+            name
+            email
+            url
+          }
+          pageInfo {
+            endCursor
+            startCursor
+          }
+          totalCount
+        }
+      }
+    }`;
+    return JSON.stringify({ query });
+  }
```

Comment:
This is a GraphQL request?

Reply:
yes, the server expects the client to send a POJO containing the `query` property, which is a graphql query string.